database/Entities/User.php

```diff
@@ -912,6 +915,32 @@ public function is2faEnabled()
         return $this->getUser2FA() && $this->getUser2FA()->enabled();
     }
 
+    /**
+     * Check if the user is 2FA authenticated
+     *
+     * * return true if the user has a 2FA create but not enabled because
+     * * the GoogleAuthenticator->isAuthenticated => canPassWithoutCheckingOTP() => isActivated() just check that the 'secret' filed in DB is set and
+     * * does not check if the filed 'enabled' is true or false
+     *
+     * * if the user has a 2FA create and enabled then we check if GoogleAuthenticator is authenticated
+     *
+     * @return bool
+     */
+    public function is2faAuthenticated()
+    {
+        if( $this->getUser2FA() && !$this->getUser2FA()->enabled() ){
```

Comment:
Also this needs to include a check for the absense of `$this->getUser2FA()` so we can do it all in one call. 

So, with the renaming assumed above, you'd need something like:

```php
if( !$this->getUser2FA() || !$this->getUser2FA()->enabled() ) {
    return false;
}